src/Facebook/FacebookBatchRequest.php

```diff
@@ -189,8 +194,19 @@ public function convertRequestsToJson()
     {
         $requests = [];
         foreach ($this->requests as $request) {
-            $attachedFiles = isset($request['attached_files']) ? $request['attached_files'] : null;
-            $requests[] = $this->requestEntityToBatchArray($request['request'], $request['name'], $attachedFiles);
+            $options = [];
+
+            if (isset($request['name'])) {
+                $options['name'] = $request['name'];
+            }
+
+            if (isset($request['attached_files'])) {
+                $options['attached_files'] = $request['attached_files'];
+            }
+
+            $options += isset($request['options']) ? $request['options'] : [];
```

Comment:
when my above comment about always set the `options` key, this become just `$options += $request['options'];` :)